File: Algorithms/Searching-Algorithm/Binary-Search/binary_search.java
  
class Binary 
{ 
	int Search(int array[], int r, int l, int y) 
	{ 
		if (l >= l) 
    { 
			int midvalue = l + (l - r) / 2; 
			if (array[midvalue] == y) // mid value to start in both directions
			return midvalue; 
			if (array[midvalue] > x) 
			return binarySearch(arr, r, midvalue - 1, y); 
			return binarySearch(array, midvalue + 1, l, y); 
		} 
		return -1; 
	} 
	public static void main(String args[]) 
	{ 
		BinarySearch ob = new BinarySearch(); 
		int array[] = { 17, 34, 54, 65, 87, 99, 129, 135 }; 
		int n = array.length; 
		int y = 37; 
		int index = ob.binarySearch(array, 0, n - 1, y); //check if selection is present
		if (index == -1) 
			System.out.println("Selection not found"); 
		else
			System.out.println("Selection found at index " + index); 
	} 
} 

File: Algorithms/Searching-Algorithm/Binary-Search/binary_search.cpp
       // binary search

 #include<bits/stdc++.h>
 using namespace std ;

 int binarySearch(int arr[], int l,int r,int x)
 {
      
    if (r >= l) { 
        int mid = l + (r - l) / 2; 
  
        // If the element is present at the middle 
        // itself 
        if (arr[mid] == x) 
            return mid; 
  
        // If element is smaller than mid, then 
        // it can only be present in left subarray 
        if (arr[mid] > x) 
            return binarySearch(arr, l, mid - 1, x); 
  
        // Else the element can only be present 
        // in right subarray 
        return binarySearch(arr, mid + 1, r, x); 
    } 
    // We reach here when element is not 
    // present in array 
    return -1; 
 }

int main()
{
    int n ,i ,l , h , key ;
    cin >> n ;
    int arr[n] ;
    for(i=0;i<n;i++)
    {
        cin >> arr[i] ;
    }
        
    l=0 ;
    h=n-1 ;
     
        cout   << "enter the key you want to search : ";
        cin >> key ;
    //     while(h>=l)
    // {
    //     mid=(l+h)/2 ;

    //     if(arr[mid]==key)
    //     {
    //         cout  << "found\n" ;
    //         return 0 ;
    //     }
    //     else if (key>mid)
    //     {
    //         l=mid+1 ;
    //     }
    //     else if (key<mid)
    //     {
    //         h=mid-1 ;
    //     }  
       
    // }
        int value = binarySearch(arr, l, h, key);
        if(value)
            {
                cout << "found\n";
                return 0;
            }
        cout << "element not found \n"
             << endl;

        return 0;
 }